plugins/modules/ios_evpn_global.py

```diff
@@ -0,0 +1,356 @@
+
+# Before state:
+# -------------
+# l2vpn evpn
+#  replication-type ingress
+#  router-id Loopback1
+#  ip local-learning disable
+#  route-target auto vni
+
+#- name: Replaces device configuration for EVPN global with provided configuration
+#  cisco.ios.ios_evpn_global:
+#    config:
+#        replication_type: static
+#        router_id: Loopback2
+#        default_gateway:
+#          advertise: true
+#        flooding_suppression:
+#          address_resolution:
+#            disable: true
+#    state: replaced
+
+# Commands Fired:
+# ---------------
+#   "commands": [
+#       "l2vpn evpn",
+#       "default-gateway advertise",
+#       "flooding-suppression address-resolution disable",
+#       "no ip local-learning disable",
+#       "replication-type static",
+#       "no route-target auto vni",
+#       "router-id Loopback2"
+#   ],
+
+# After state:
+# ------------
+# l2vpn evpn
+#  replication-type static
+#  flooding-suppression address-resolution disable
+#  router-id Loopback2
+#  default-gateway advertise
+
+# Using state Deleted
+
+# Before state:
+# -------------
+# l2vpn evpn
+#  replication-type static
+#  flooding-suppression address-resolution disable
+#  router-id Loopback2
+#  default-gateway advertise
+
+# - name: Delete EVPN global
+#   cisco.ios.ios_evpn_global:
+#     config:
+#     state: deleted
+
+# Commands Fired:
+# ---------------
+#  "commands": [
+#      "no l2vpn evpn"
+#      ]
+
+# After state:
+# -------------
+#
+# Using gathered
+
+# Before state:
+# -------------
+#
+# l2vpn evpn
+#  replication-type ingress
+#  router-id Loopback1
+#  ip local-learning disable
+#  route-target auto vni
+
+# - name: Gather facts for evpn_global
+#   cisco.ios.ios_evpn_global:
+#     config:
+#     state: gathered
+
+# Task Output:
+# ------------
+#
+# gathered:
+#   replication_type: ingress
+#   route_target:
+#     auto:
+#       vni: true
+#   router_id: Loopback1
+#   ip:
+#     local_learning:
+#       disable: true
+
+# Using Rendered
+
+# - name: Rendered the provided configuration with the existing running configuration
+#   cisco.ios.ios_evpn_global:
+#     config:
+#         replication_type: static
+#         route_target:
+#           auto:
+#             vni: true
+#     state: rendered
+
+# Task Output:
+# ------------
+#
+# rendered:
+# - l2vpn evpn
+# - replication-type static
+# - route-target auto vni
+
+# Using parsed
+
+# File: parsed.cfg
+# ----------------
+#
+# l2vpn evpn
+#  replication-type ingress
+#  router-id Loopback1
+#  default-gateway advertise
+#  route-target auto vni
+
+# - name: Parse the commands for provided configuration
+#   cisco.ios.ios_evpn_global:
+#     running_config: "{{ lookup('file', 'parsed.cfg') }}"
+#     state: parsed
+
+# Task Output:
+# ------------
+#
+# parsed:
+#     replication_type: ingress
+#     route_target:
+#       auto:
+#         vni: true
+#     router_id: Loopback1
+#     default_gateway:
+#       advertise: true
+
+"""
+
+RETURN = """
+before:
```

Comment:
The return attributes contain more attributes 
please refer to https://github.com/ansible-collections/cisco.ios/blob/56b92e89d0bf2a7c36772279ca6c4fe963859ada/plugins/modules/ios_interfaces.py#L926 for all the options.

Reply:
Corrected